graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/db/PaginatedPipelineService.java

```diff
@@ -16,34 +16,11 @@
  */
 package org.graylog.plugins.pipelineprocessor.db;
 
-import org.graylog2.bindings.providers.MongoJackObjectMapperProvider;
-import org.graylog2.database.MongoConnection;
-import org.graylog2.database.PaginatedDbService;
 import org.graylog2.database.PaginatedList;
 import org.graylog2.search.SearchQuery;
-import org.mongojack.DBQuery;
-import org.mongojack.DBSort;
 
-import javax.inject.Inject;
 import java.util.function.Predicate;
 
-public class PaginatedPipelineService extends PaginatedDbService<PipelineDao> {
-    private static final String COLLECTION_NAME = "pipeline_processor_pipelines";
-
-    @Inject
-    public PaginatedPipelineService(MongoConnection mongoConnection,
-                                     MongoJackObjectMapperProvider mapper)
-    {
-        super(mongoConnection, mapper, PipelineDao.class, COLLECTION_NAME);
-    }
-
-    public long count() {
-        return db.count();
-    }
-
-    public PaginatedList<PipelineDao> findPaginated(SearchQuery searchQuery, Predicate<PipelineDao> filter, int page, int perPage, String sortField, String order) {
-        final DBQuery.Query dbQuery = searchQuery.toDBQuery();
-        final DBSort.SortBuilder sortBuilder = getSortBuilder(order, sortField);
-        return findPaginatedWithQueryFilterAndSort(dbQuery, filter, sortBuilder, page, perPage);
-    }
+public interface PaginatedPipelineService {
```

Comment:
Since this contains an apparent breaking Java code API change (from concrete implementation to interface), I think it would be good to describe that in the https://github.com/Graylog2/graylog2-server/blob/f3fec9422a5190ee61094574c70cfe6c23928d74/UPGRADING.md file, so that users will have a heads up, in cases they have extended or used those classes in some way.

Reply:
Ok, good call. I updated the UPGRADING.md file. I tried to match the format but wasn't sure if there is a preferred way to notate changes like these, so let me know if you think I need to make any changes to the note.